### DISCUSSION TITLE
API operating problem

### DISCUSSION THREAD

**Discussion by CogalTek** [pocketbase/pocketbase#3582] [Q&A] [answered]
Hello, I am trying to make the api work with javascript, I created the pb_hooks/ folder in the same directory as pb_data, I added a main.pb.js file and I added this code:

```Javascript
// pb_hooks/main.pb.js

routerAdd("GET", "/hello", (c) => {
    return c.json(200, { "message": "Hello" })
})

```

when I make my request I get:
```
{
    "code": 404,
    "message": "Not Found.",
    "data": {}
}
```

Does anyone have any idea why this isn't working?

**Comment by ganigeorgiev** [comment]
How do you send the request? Could you provide a code sample?

For sending requests to custom routes using the JS SDK, you can also check https://pocketbase.io/docs/js-routing/#sending-request-to-custom-routes-using-the-sdks.

**Reply by CogalTek** [reply]
with postman 
<img width="1080" alt="Capture d’écran 2023-10-22 à 16 08 16" src="https://github.com/pocketbase/pocketbase/assets/91875582/6cae2519-8c5b-49e9-9798-132d27247633">

**Reply by ganigeorgiev** [reply]
The path is just `/hello` in your case.

We don't automatically prefix custom routes.

**Reply by CogalTek** [reply]
like that ? 

<img width="1080" alt="Capture d’écran 2023-10-22 à 16 09 31" src="https://github.com/pocketbase/pocketbase/assets/91875582/d00a0fb7-fbb4-45d8-b5a4-2114e6348dbd">

**Reply by ganigeorgiev** [reply, marked as answer]
Yes. If the hook file and directory were created after the application was started you'll have to restart the process as there were no files to watch.
